### PR TITLE
Fleet status tests

### PR DESCRIPTION
**Pull request by zeisss**
- Replaces the custom mock with a library which also provides to define number of calls, not calls etc.
- Add test for simple successfull `fleet.GetStatus` call
- Refactors `fleet` to have a `GetStatusWithMatcher` func so we don't need a regexp everywhere

Will have merge conflicts after https://github.com/giantswarm/formica/pull/25

### CODE REVIEW

**Comment by zeisss** [comment]
ping @giantswarm/cena

**Comment by zeisss** [comment]
I would like to merge #27 beforehand

**Comment by xh3b4sd** [comment]
I have no strong feelings on this.

File: fleet/fleet.go [hunk 1]
```diff
@@ -234,6 +232,13 @@ func (f fleet) GetStatus(name string) (UnitStatus, error) {
 }
 
 func (f fleet) GetStatusWithExpression(exp *regexp.Regexp) ([]UnitStatus, error) {
+	status, err := f.GetStatusWithMatcher(exp.MatchString)
+	return status, maskAny(err)
+}
+
+// GetStatusWithMatcher returns a UnitStatus for each unit where the given matcher
```
Comment (JosephSalisbury) [inline fleet/fleet.go:239]: This is slightly misleading, if I understand correctly, as we can return `[]UnitStatus`, not just `UnitStatus`

Comment (JosephSalisbury) [inline fleet/fleet.go:239]: Well, not misleading, maybe it's just me - I'd go with `GetStatusWithMatcher returns a list of UnitStatus for each unit where the given matcher returns true`

**Comment by JosephSalisbury** [comment]
Yup, LGTM excluding small documentation comment

File: fleet/fleet.go (hunk 1 above)
Comment (zeisss) [inline fleet/fleet.go:239]: doesn't the "for each unit" imply that it is a list?

Comment (JosephSalisbury) [inline fleet/fleet.go:239]: Hence my hesitance :D - I think when I quickly scan it, I see `returns a UnitStatus`, and get slightly confused

**Comment by zeisss** [comment]
Due to rebase and merges this ended up as a86f95686571534f202b4bdef724e86a793876c1 in master.

**Pull request closed without merging**